File: GYTableViewController/GYTableViewController/Demo/RefreshDemo/RefreshHotViewCell.m
//
//  RefreshHotViewCell.m
//  GYTableViewController
//
//  Created by 高扬 on 2018/3/24.
//  Copyright © 2018年 高扬. All rights reserved.
//

#import "RefreshHotViewCell.h"
#import "HotModel.h"

@interface HomeFastItem : UIControl

@property (nonatomic, strong) UILabel *titleLable;
@property (nonatomic, strong) UIImageView *iconView;//图片文本
@property (nonatomic, strong) HotModel *hotModel;

@end

@implementation HomeFastItem

- (void)layoutSubviews {
    [super layoutSubviews];
    
    self.titleLable.text = self.hotModel.title;
    [self.titleLable sizeToFit];
    
    self.iconView.image = [UIImage imageNamed:self.hotModel.iconName];
    CGFloat const nodeSizeWidth = 40;
    self.iconView.size = CGSizeMake(nodeSizeWidth,nodeSizeWidth);
    
    self.titleLable.centerX = self.iconView.centerX = self.width / 2.;
    CGFloat const gap = 10;
    CGFloat const baseY = (self.height - self.titleLable.height - self.iconView.height - gap) / 2;
    self.iconView.y = baseY;
    self.titleLable.y = self.iconView.maxY + gap;
}

- (UILabel *)titleLable {
    if (!_titleLable) {
        _titleLable = [UICreationUtils createLabel:TVStyle.sizeTextPrimary color:TVStyle.colorTextPrimary];
        [self addSubview:_titleLable];
    }
    return _titleLable;
}

- (UIImageView *)iconView {
    if (!_iconView) {
        _iconView = [[UIImageView alloc]init];
        [self addSubview:_iconView];
    }
    return _iconView;
}

@end

@interface RefreshHotViewCell()

@property (nonatomic, strong)UIView *bottomLine;

@end

@implementation RefreshHotViewCell

#pragma mark 根据外部传入数据开始布局
/**
 * 批量创建HomeFastItem按钮实例，一字排开
 */
- (void)showSubviews {
    self.backgroundColor = [UIColor whiteColor];
    
    NSArray<HotModel *> *hotModels = [self getCellData];
    CGFloat const itemWidth = self.contentView.width / hotModels.count;
    for (NSInteger i = 0; i < hotModels.count; i++) {
        HotModel* hotModel = hotModels[i];
        HomeFastItem* item = [[HomeFastItem alloc]init];
        item.hotModel = hotModel;
        [self.contentView addSubview:item];
        item.frame = CGRectMake(i * itemWidth, 0, itemWidth, self.contentView.height);
//        item.tag = i;
    }
    
    self.bottomLine.x = 0;
    self.bottomLine.maxY = self.contentView.height;
    self.bottomLine.width = self.contentView.width;
}

- (UIView *)bottomLine {
    if (!_bottomLine) {
        _bottomLine = [[UIView alloc]init];
        _bottomLine.backgroundColor = TVStyle.colorLine;
        _bottomLine.height = TVStyle.lineWidth;
        [self.contentView addSubview:_bottomLine];
    }
    return _bottomLine;
}

@end

File: GYTableViewController4Swift/GYTableViewController4Swift/Demo/RefreshDemo/RefreshHotViewCell.swift
//
//  RefreshHotViewCell.swift
//  GYTableViewController4Swift
//
//  Created by gaoyang on 2019/2/3.
//  Copyright © 2019年 高扬. All rights reserved.
//

import UIKit

class HomeFastItem: UIControl {
    
    var hotModel:HotModel?;
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        if self.hotModel == nil {
            return
        }
        
        self.titleLable.text = self.hotModel?.title;
        self.titleLable.sizeToFit();
        
        self.iconView.image = UIImage.init(named: self.hotModel!.iconName);
        let nodeSizeWidth:CGFloat = 40;
        self.iconView.size = CGSize.init(width: nodeSizeWidth, height: nodeSizeWidth)
        
        self.titleLable.centerX = self.width / 2.0;
        self.iconView.centerX = self.titleLable.centerX ;
        let gap:CGFloat = 10;
        let baseY:CGFloat = (self.height - self.titleLable.height - self.iconView.height - gap) / 2.0;
        self.iconView.y = baseY;
        self.titleLable.y = self.iconView.maxY + gap;
    }
    
    private lazy var titleLable:UILabel = { () -> UILabel in
        let _titleLable:UILabel = UICreationUtils.createLabel(TVStyle.sizeTextPrimary, color: TVStyle.colorTextPrimary)
        self.addSubview(_titleLable)
        return _titleLable
    }()
        
    private lazy var iconView:UIImageView = { () -> UIImageView in
        let _iconView = UIImageView()
        self.addSubview(_iconView)
        return _iconView;
    }()
    
}

class RefreshHotViewCell: GYTableViewCell {
    
    /**
     * 批量创建HomeFastItem按钮实例，一字排开
     */
    override func showSubviews() {
        super.showSubviews()
        
        self.backgroundColor = UIColor.white;
        
        let hotModels:Array<HotModel> = self.getData() as! Array<HotModel>
        let itemWidth:CGFloat = self.contentView.width / CGFloat(hotModels.count);
        
        for i in 0..<hotModels.count {
            let hotModel:HotModel = hotModels[i];
            let item:HomeFastItem = HomeFastItem.init(frame: CGRect.zero);
            item.hotModel = hotModel;
            self.contentView.addSubview(item);
            item.frame = CGRect.init(x: CGFloat(i) * itemWidth, y: 0, width: itemWidth, height: self.contentView.height)
            //        item.tag = i;
        }
        
        self.bottomLine.x = 0;
        self.bottomLine.maxY = self.contentView.height;
        self.bottomLine.width = self.contentView.width;
    }
    
    private lazy var bottomLine:UIView = { () -> UIView in
        let _bottomLine = UIView()
        _bottomLine.backgroundColor = TVStyle.colorLine;
        _bottomLine.height = TVStyle.lineWidth;
        self.contentView.addSubview(_bottomLine);
        return _bottomLine
    }()

}
